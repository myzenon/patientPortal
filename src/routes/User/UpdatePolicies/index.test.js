import React from 'react'
import { UpdatePolicies } from './index'
import ReactGA from 'react-ga';

   // This should be part of your setup
   ReactGA.initialize('UA-121783543-1', {
       'debug': true
     });;

jest.mock("react-ga")

jest.mock('assets/policy/nopp.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\nnopp content')
jest.mock('assets/policy/eula.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\neula content')
jest.mock('assets/policy/terms.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\nterms content')
jest.mock('assets/policy/access.html', () => '<meta charset="UTF-8">\n<meta version="v1.0">\naccess content')

test('renders correctly', () => {
    const wrapper = render (
        <UpdatePolicies />
    )
    expect(toJson(wrapper)).toMatchSnapshot()
})

test('checkbox checked | next button enabled', () => {
    const wrapper = mount (
        <UpdatePolicies />
    )
    wrapper.find('input[type="checkbox"]').simulate('change')
    expect(toJson(wrapper.find('button').render())).toMatchSnapshot()
})

test('when user accept | usersUpdateUserPut called | move to comfirm info page', done => {
    const props = {
        patient: {
            ...mockObj.patient,
            noppStatus: false
        },
        callAPIGateway: (name, get, post) => {
            const date = new Date()
            expect(name).toBe('usersUpdateUserPut')
            expect(post).toEqual({
                'nopp': {
                    'Date': date.getFullYear() + '-' + ('0' + (date.getMonth() + 1)).slice(-2) + '-' + date.getDate(),
                    'vEnd': 'v1.0',
                    'vNopp': 'v1.0',
                    'vTerms': 'v1.0'
                },
                'noppStatus': true
            })
            return Promise.resolve()
        },
        setPatient: (patient) => {
            expect(patient).toEqual({
                ...mockObj.patient,
                noppStatus: true
            })
        },
        push: (url) => {
            expect(url).toBe('/confirm-info')
            done()
        }
    }
    const wrapper = mount (
        <UpdatePolicies {...props} />
    )
    setTimeout(() => {
        wrapper.find('input[type="checkbox"]').simulate('change')
        wrapper.update()
        wrapper.find('button[type="button"]').simulate('click')
    }, 0)
})