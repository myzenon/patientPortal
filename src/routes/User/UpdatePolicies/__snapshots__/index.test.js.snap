// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkbox checked | next button enabled 1`] = `
<button
  class="button btn btn-secondary"
  type="button"
>
  NEXT
</button>
`;

exports[`renders correctly 1`] = `
<div
  class="container-fluid"
>
  <div
    class="content container"
  >
    <p>
      <strong>
        We at Arine take your Privacy Very Seriously.
      </strong>
    </p>
    <p>
      Our Policies have changed. Please read the statements below about how medical information about you may be used and disclosed and how you can get access to this information.
    </p>
    <div
      class="policies"
    >
      <div
        class="wrapperNoPopup"
      >
        <div
          class="policyWrapper"
        >
          <div
            class="policyContent"
          >
            <div
              class="tabWrapper"
            >
              <div
                class="tabActive"
              >
                Notice of Privacy Practices
              </div>
              <div
                class="tab"
              >
                End User License Agreement
              </div>
              <div
                class="tab"
              >
                Terms and Conditions
              </div>
              <div
                class="tabSpace"
              />
            </div>
            <div
              class="contentWrapper"
            >
              <div
                class="scrollDiv"
              >
                <span>
                  <meta
                    charset="UTF-8"
                  />
                  

                  <meta
                    version="v1.0"
                  />
                  
nopp content
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="agreeCheckbox"
    >
      <div
        class="row"
      >
        <div
          class="col-2"
        >
          <input
            type="checkbox"
            value="false"
          />
        </div>
        <div
          class="col-10"
        >
          I have read and agree to the Notice of Privacy Practices, Terms of Service and End User License Agreement
        </div>
      </div>
    </div>
    <div>
      <button
        class="button btn btn-secondary disabled"
        disabled=""
        type="button"
      >
        NEXT
      </button>
    </div>
    <div
      class="questionWrapper"
    >
      Questions ?
    </div>
  </div>
</div>
`;
